client/src/components/History/CurrentCollection/CollectionNavigation.vue

```diff
@@ -12,6 +20,7 @@
 <script>
 export default {
     props: {
+        history: { type: Object, required: true },
```

Comment:
It's better to only use simple prop types (whenever possible), like e.g. String, Boolean and so on. The props are reactive and here, parsing historyName as String seems to be sufficient. This also makes it easier to see which attribute is actually needed within the component.